cpp/FileTransferAgent.cpp

```diff
@@ -222,6 +223,11 @@ void Snowflake::Client::FileTransferAgent::upload(string *command)
         renewToken(command);
         i--;
       }
+      else if( outcome == RemoteStorageRequestOutcome::FAILED)
+      {
+        throw SnowflakeTransferException(TransferError::INTERNAL_ERROR,
```

Comment:
Can you improve this message ? What is the file we failed to upload, and why did it fail ?

Reply:
And what about reporting errors and checking for outcomes on small files ?

Reply:
@sfc-gh-sshankar For small files its done on line 272.
We did add at multiple locations the reason for error. 
Will add more detail to the exception.